### PR TITLE
Add macos tripwire file

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Unconditionally touch a file during the macos_assemble script. Depending on this file as an input allows us to bypass Xcode's build caching and rely on the flutter tool build caching.

This is a requirement to provide the xcfilelist inputs and outputs to xcode, which are then used to ensure the build phases are executed in the correct order.

### CODE REVIEW

File: packages/flutter_tools/bin/macos_assemble.sh [hunk 1]
```diff
@@ -62,6 +62,12 @@ ephemeral_dir="${SOURCE_ROOT}/Flutter/ephemeral"
 build_inputs_path="${ephemeral_dir}/FlutterInputs.xcfilelist"
 build_outputs_path="${ephemeral_dir}/FlutterOutputs.xcfilelist"
 
+# Unconditionally touch a tripwire file. This can be listed as an input to
+# ensure that the flutter tool always has a chance to use its own cache.
+# Removeing this file as an input requires a separate xcode scheme per
```
Comment (stuartmorgan) [inline packages/flutter_tools/bin/macos_assemble.sh]: Nit: removing

**Review by stuartmorgan-g — COMMENTED** [review]
Why isn't this being added within assemble itself? I was thinking that assemble could do this step, and also add it as an input in the file list, and then the project template doesn't need to know about any of this.

**Comment by jonahwilliams** [comment]
Then it would be listed as an output, and wouldn't trip xcodes timestamp analysis

**Comment by stuartmorgan** [comment]
I don't follow; why can't we have `assemble` add it to FlutterInputs.xcfilelist (not FlutterOutputs.xcfilelist)?

**Comment by jonahwilliams** [comment]
Well, hypothetically with this done in macos_assemble.sh, someone could still setup and xcode project with multiple schemes in a such a way that the timestamp analysis was correct without the tripwire. If I added to the assemble inputs, then it force everyone to not use it. This is a workaround that is very specific to this particular incarnation of the xcode project.

**Comment by stuartmorgan** [comment]
> Well, hypothetically with this done in macos_assemble.sh, someone could still setup and xcode project with multiple schemes in a such a way that the timestamp analysis was correct without the tripwire.

It seems like getting the right FlutterInputs.xcfilelist would be quite difficult, since someone would need to generate them for each config and then copy them to a known location, right? (And then repeat that any time they updated Flutter..)

> If I added to the assemble inputs, then it force everyone to not use it. This is a workaround that is very specific to this particular incarnation of the xcode project.

Okay, well if we want it to be specific to the Xcode project then let's just move it fully back into the project and close this. Splitting it between the two seem like it makes it more confusing.

When moving it back, you can just make it another line at the start of the existing script step, rather than making a new script step, which will make things much simpler than the earlier version of the change.

**Comment by jonahwilliams** [comment]
Sounds good

**Pull request closed without merging**